ui/apps/platform/src/Containers/Vulnerabilities/NodeCves/NodeCve/AffectedNodesTable.tsx

```diff
@@ -0,0 +1,207 @@
+import { gql } from '@apollo/client';
+import { Link } from 'react-router-dom';
+
+import {
+    TbodyLoading,
+    TbodyError,
+    TbodyEmpty,
+    TbodyFilteredEmpty,
+} from 'Components/TableStateTemplates';
+import { TableUIState } from 'utils/getTableUIState';
+
+import useSet from 'hooks/useSet';
+import { VulnerabilitySeverity, isVulnerabilitySeverity } from 'types/cve.proto';
+import { severityRankings } from 'constants/vulnerabilities';
+import VulnerabilitySeverityIconText from 'Components/PatternFly/IconText/VulnerabilitySeverityIconText';
+import VulnerabilityFixableIconText from 'Components/PatternFly/IconText/VulnerabilityFixableIconText';
+
+import CvssFormatted from '../../components/CvssFormatted';
+import { getNodeEntityPagePath } from '../../utils/searchUtils';
+import NodeComponentsTable from './NodeComponentsTable';
+
+export const affectedNodeFragment = gql`
+    fragment AffectedNode on Node {
+        id
+        name
+        operatingSystem
+        cluster {
+            name
+        }
+        nodeComponents {
+            name
+            version
+            source
+            nodeVulnerabilities {
+                vulnerabilityId: id
+                cve
+                severity
+                fixedByVersion
+                cvss
+                scoreVersion
+            }
+        }
+    }
+`;
+
+export type AffectedNode = {
+    id: string;
+    name: string;
+    operatingSystem: string;
+    cluster: {
+        name: string;
+    };
+    nodeComponents: {
+        name: string;
+        version: string;
+        source: string;
+        nodeVulnerabilities: {
+            vulnerabilityId: string;
+            cve: string;
+            severity: string;
+            fixedByVersion: string;
+            cvss: number;
+            scoreVersion: string;
+        }[];
+    }[];
+};
+
+type NodeVulnerabilities = AffectedNode['nodeComponents'][0]['nodeVulnerabilities'];
+
+function getHighestVulnerabilitySeverity(vulns: NodeVulnerabilities): VulnerabilitySeverity {
+    let topSeverity: VulnerabilitySeverity = 'UNKNOWN_VULNERABILITY_SEVERITY';
+    vulns.forEach(({ severity }) => {
+        if (
+            isVulnerabilitySeverity(severity) &&
+            severityRankings[severity] > severityRankings[topSeverity]
+        ) {
+            topSeverity = severity;
+        }
+    });
+    return topSeverity;
+}
+
+function getAnyVulnerabilityIsFixable(vulns: NodeVulnerabilities): boolean {
+    return vulns.some((vuln) => vuln.fixedByVersion !== '');
+}
+
+function getHighestCvssScore(vulns: NodeVulnerabilities): {
+    cvss: number;
+    scoreVersion: string;
+} {
+    let topCvss = 0;
+    let topScoreVersion = 'N/A';
+    vulns.forEach(({ cvss, scoreVersion }) => {
+        if (cvss > topCvss) {
+            topCvss = cvss;
+            topScoreVersion = scoreVersion;
+        }
+    });
+    return { cvss: topCvss, scoreVersion: topScoreVersion };
+}
```

Comment:
These are all slightly different versions of utils in WorkloadCVEs. It might be worth a follow up to abstract them into common functions, but it would be an abstraction for only two use cases at this point. (At least up to the point we've planned for VM that I am aware of.)